discretize/base/base_tensor_mesh.py

```diff
@@ -996,6 +990,346 @@ def innerProductDeriv(v=None):
         else:
             return None
 
+
+    def _fastFacePropertiesInnerProduct(
+        self, projection_type, model=None, invert_model=False, invert_matrix=False
+    ):
+        """Fast version of get_face_inner_product_deriv.
+
+        This does not handle the case of a full tensor property.
+
+        Parameters
+        ----------
+        projection_type : str
+            'edges' or 'faces'
+        model : numpy.ndarray
+            material property (tensor properties are possible) at each cell center (nC, (1, 3, or 6))
+        invert_model : bool
+            inverts the material property
+        invert_matrix : bool
+            inverts the matrix
+
+        Returns
+        -------
+        (n_faces, n_faces) or (n_edges, n_edges) scipy.sparse.csr_matrix
+            M, the inner product matrix
+
+        """
+        projection_type = projection_type[0].upper()
+        if projection_type not in ["F", "E"]:
+            raise ValueError("projection_type must be 'F' for faces or 'E' for edges")
+
+        if model is None:
+            model = np.ones(self.nF)
+
+        if invert_model:
+            model = 1.0 / model
+
+        if is_scalar(model):
+            model = model * np.ones(self.nF)
+        # COULD ADD THIS CASE IF DESIRED
+        # elif len(model) == self.dim:
+        #     model = np.r_[
+        #         [model[ii]*self.vnF[ii] for ii in range(0, self.dim)]
+        #     ]
+
+        # number of elements we are averaging (equals dim for regular
+        # meshes, but for cyl, where we use symmetry, it is 1 for edge
+        # variables and 2 for face variables)
+        if self._meshType == "CYL":
+            shape = getattr(self, "vn" + projection_type)
+            if self.is_symmetric:
+                n_elements = 1
+            else:
+                n_elements = sum([1 if x != 0 else 0 for x in shape]) - 1
+        else:
+            n_elements = self.dim - 1
+
+        # Isotropic case only
+        if model.size == self.nF:
+            Aprop = self.face_areas * mkvc(model)
+            if projection_type == 'E':
+                Av = getattr(self, "average_edge_to_face")
+                M = n_elements * sdiag(Av.T * Aprop)
```

Comment:
I think we might need to be a bit careful here. When we take the transpose of this operator, 

```
mesh = discretize.TensorMesh([2, 1, 1])
A = mesh.average_edge_to_face.todense()
```

If we take the sum across all columns, they should sum to 1 (e.g. this is a True average) 
```
A.sum(1)
```
which it does. 

But, if we take the transpose, the columns don't sum to 1 
```
A.T.sum(1) 
```
![image](https://github.com/simpeg/discretize/assets/6361812/5476201f-cecc-461b-ad37-e9ab17c31b1c)


I think what we should be doing is define 
```
Ae2f = mesh.average_edge_to_face.todense()
w = Ae2f.T.sum(1)
Af2e = discretize.utils.sdiag(1./w) * Ae2f.T
```

Reply:
nevermind! Thanks for straightening this out for me @jcapriot. Conceptually, we think about averaging the field to cell centers and then taking the product